tests/basic/pthread_detach/README.md

```diff
@@ -0,0 +1,181 @@
+# Preliminaries
+## Thread identities
+A thread has 3 identities:
+- Linux process context - `task_struct`
+- LKL arch specific process context - `thread_info`
+- LKL host thread context - `lthread`
+
+For most of a thread's lifetime these 3 identities should have a 1:1 mapping between them.
+One instance where this doesn't hold is on the return path of the clone() system call. (This is covered in more detail later in the `pthread_create` section.)
+
+## LKL's CPU Lock: lkl_cpu_get/lkl_cpu_put
+The owner of the lock is assigned and checked based on the lthread identity.
+
+## sched_sem: thread level semaphore LKL uses to control scheduling
+From the LKL paper -
+> To regain control of scheduling, the generic LKL architecture layer associates
+> an environment-provided semaphore with each LKL thread.
+>
+> Immediately after creation, and before running any Linux code each LKL thread
+> acquires its corresponding semaphore, and gets blocked as the semaphore's
+> initial value is 0.
+>
+> When the Linux scheduler selects a new thread to run, it releases the semaphore
+> of the new thread and immediately acquires its own semaphore.
+>
+> The new thread will start running and the old one stops.
+>
+> This token passing mechanism ensures that at any given time there is only one
+> thread running and the scheduling order is dictated by the Linux scheduler.
+
+O. Purdila, L. A. Grijincu and N. Tapus, "LKL: The Linux kernel library," 9th RoEduNet IEEE International Conference, Sibiu, 2010, pp. 328-333.
+
+The environment-provided semaphore is the `sched_sem` field in thread_info.
+
+## thread_sched_jb()
+`thread_sched_jb` is LKL's way of doing co-operative multitasking, by yielding to the scheduler. Its invoked from - 
+
+1. `lkl_cpu_put`: If the Linux scheduler run queue has >= 1 tasks.
+2. `switch_to_host_task(task)`: If `current` process context != task.
+3. `lkl_syscall`: If syscall is NR_REBOOT, yields to scheduler after running the syscall.
+
```

Comment:
Document how `thread_sched_jb` exits?  That's the most confusing thing here.

Reply:
Have documented how `switch_to_host_task` usage of this function returns in the next section. Let me look up how the other two usages exit.

Reply:
Done.